docs/security-considerations.rst

```diff
@@ -273,17 +290,17 @@ control to a fixed and trusted third party or just converts the contract into
 a simple "give me back my money" contract.
 
 
-*******************
-Formal Verification
-*******************
+Method visibility from mother contracts and ABI
+===============================================
+
+
+When compiling a contract that inherits from other contracts, the generated JSON ABI for the child contract does not contain all the methods from contracts in the upper hierarchy. However, that does not prevent public methods from being executed. You should take care of which methods can be executed. Use `internal` function modifier to control visibility.
```

Comment:
Maybe the actual problem is libraries...see this explanation of the Parity multisig attack:

https://blog.zeppelin.solutions/on-the-parity-wallet-multisig-hack-405a8c12e8f7

There was a public function in the library, and that was reachable because the fallback in the main contract forwarded unmatched function calls to the library. In that scenario it makes more sense that the ABI wouldn't include the function; the contract itself doesn't recognize it either, otherwise the call wouldn't have gone through the fallback.